### ISSUE TITLE
pyright unable to type-check Union type with nested TypeVars

### ISSUE THREAD

**Issue by mpereira** [microsoft/pyright#1066]
**To Reproduce**

`pyright_test.py`
```
from typing import Generic, TypeVar, Union, Final, final, Optional

T = TypeVar("T")  # Success type.
E = TypeVar("E")  # Error type.


@final
class Ok(Generic[T]):
    def __init__(self, value: T) -> None:
        self._value: Final = value


@final
class Err(Generic[E]):
    def __init__(self, value: E) -> None:
        self._value: Final = value


Result = Union[Ok[T], Err[E]]


def return_ok_none() -> Result[Optional[int], Exception]:
    return Ok(None)


def return_ok_one() -> Result[Optional[int], Exception]:
    return Ok(1)
```

```
$ mypy --version
mypy 0.782
$ mypy pyright_test.py
Success: no issues found in 1 source file
```

```
$ pyright --version
pyright 1.1.74
$ pyright pyright_test.py
No configuration file found.
stubPath typings is not a valid directory.
Assuming Python platform Darwin
Searching for source files
Found 1 source file
<...>/pyright_test.py
  22:12 - error: Expression of type "Ok[None]" cannot be assigned to return type "Result[int | None, Exception]"
    Type "Ok[None]" cannot be assigned to type "Result[int | None, Exception]"
      "Ok[None]" is incompatible with "Err[Exception]"
        TypeVar "T" is invariant
          Type "None" cannot be assigned to type "int | None" (reportGeneralTypeIssues)
  25:12 - error: Expression of type "Ok[Literal[1]]" cannot be assigned to return type "Result[int | None, Exception]"
    Type "Ok[Literal[1]]" cannot be assigned to type "Result[int | None, Exception]"
      "Ok[Literal[1]]" is incompatible with "Err[Exception]"
        TypeVar "T" is invariant
          Type "Literal[1]" cannot be assigned to type "int | None" (reportGeneralTypeIssues)
2 errors, 0 warnings, 0 infos 
Completed in 0.505sec
```

**Expected behavior**
I would expect this type check to be successful like in mypy.

**Screenshots or Code**

See "To reproduce" section.

**Additional context**
```
$ python3 --version
Python 3.8.5
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by mpereira** [comment]
Thanks for the quick triaging!

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.75, which I just published. It will also be included in the next version of Pylance.

**Comment by mpereira** [comment]
Just tested it on the codebase I first noticed the issue, works great. Thank you!
